unixfs/src/pb/unixfs.rs

```diff
@@ -0,0 +1,229 @@
+            + self
+                .blocksizes
+                .iter()
+                .map(|s| 1 + sizeof_varint(*(s) as u64))
+                .sum::<usize>()
+            + self
+                .hashType
+                .as_ref()
+                .map_or(0, |m| 1 + sizeof_varint(*(m) as u64))
+            + self
+                .fanout
+                .as_ref()
+                .map_or(0, |m| 1 + sizeof_varint(*(m) as u64))
+            + self
+                .mode
+                .as_ref()
+                .map_or(0, |m| 1 + sizeof_varint(*(m) as u64))
+            + self
+                .mtime
+                .as_ref()
+                .map_or(0, |m| 1 + sizeof_len((m).get_size()))
+    }
+    fn write_message<W: WriterBackend>(&self, w: &mut Writer<W>) -> Result<()> {
+        w.write_with_tag(8, |w| w.write_enum(*&self.Type as i32))?;
+        if let Some(ref s) = self.Data {
+            w.write_with_tag(18, |w| w.write_bytes(&**s))?;
+        }
+        if let Some(ref s) = self.filesize {
+            w.write_with_tag(24, |w| w.write_uint64(*s))?;
+        }
+        for s in &self.blocksizes {
+            w.write_with_tag(32, |w| w.write_uint64(*s))?;
+        }
+        if let Some(ref s) = self.hashType {
+            w.write_with_tag(40, |w| w.write_uint64(*s))?;
+        }
+        if let Some(ref s) = self.fanout {
+            w.write_with_tag(48, |w| w.write_uint64(*s))?;
+        }
+        if let Some(ref s) = self.mode {
+            w.write_with_tag(56, |w| w.write_uint32(*s))?;
+        }
+        if let Some(ref s) = self.mtime {
+            w.write_with_tag(66, |w| w.write_message(s))?;
+        }
+        Ok(())
+    }
+}
+pub mod mod_Data {
+    #[derive(Debug, PartialEq, Eq, Clone, Copy)]
+    pub enum DataType {
+        Raw = 0,
+        Directory = 1,
+        File = 2,
+        Metadata = 3,
+        Symlink = 4,
+        HAMTShard = 5,
+    }
+    impl Default for DataType {
+        fn default() -> Self {
+            DataType::Raw
+        }
+    }
+    impl From<i32> for DataType {
+        fn from(i: i32) -> Self {
+            match i {
+                0 => DataType::Raw,
+                1 => DataType::Directory,
+                2 => DataType::File,
+                3 => DataType::Metadata,
+                4 => DataType::Symlink,
+                5 => DataType::HAMTShard,
+                _ => Self::default(),
+            }
+        }
+    }
+    impl<'a> From<&'a str> for DataType {
+        fn from(s: &'a str) -> Self {
+            match s {
+                "Raw" => DataType::Raw,
+                "Directory" => DataType::Directory,
+                "File" => DataType::File,
+                "Metadata" => DataType::Metadata,
+                "Symlink" => DataType::Symlink,
+                "HAMTShard" => DataType::HAMTShard,
+                _ => Self::default(),
+            }
+        }
+    }
+    impl From<DataType> for i32 {
```

Comment:
can't `DataType` just be cast directly to `i32`?

Reply:
I wonder if that would require `repr(i32)`? Haven't ever used it and always written similar conversions to this :)

Reply:
Nah, it just works with a straight `as i32` 👌